Exerscise_3/main.cpp

```diff
@@ -70,7 +71,7 @@ std::ifstream read_from_file(const std::string &file) {
 
 
 int main(int args, char **argv){
```

Comment:
The main is to long. Please, divide it into smaller functions. Ideally, the whole main body would be a call to a function, which parses the arguments and do specific actions.